### PR TITLE
Fix same parameter generating bug in TPE

### PR DESCRIPTION
**Pull request by leelaylay**

### CODE REVIEW

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py [hunk 1]
```diff
@@ -258,14 +249,14 @@ def receive_trial_result(self, parameter_id, parameters, value):
         trial['state'] = hp.JOB_STATE_DONE
         trials.insert_trial_docs([trial])
         trials.refresh()
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: release memory of queue?

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py [hunk 2]
```diff
@@ -286,3 +277,26 @@ def miscs_update_idxs_vals(self, miscs, idxs, vals,
...
+    
+    def update_parameters_queue(self):
+        '''update the parameters queue when it is empty or when one trial result is received
+        '''
+        rval = self.rval
+        trials = rval.trials
+        algorithm = rval.algo
+        new_ids = rval.trials.new_trial_ids(self.max_queue_len)
+        rval.trials.refresh()
+        random_state = rval.rstate.randint(2**31-1)
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: does 2**31 will cost time?

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py [hunk 3]
```diff
@@ -286,3 +277,26 @@ def miscs_update_idxs_vals(self, miscs, idxs, vals,
...
+        new_trials = algorithm(new_ids, rval.domain, trials, random_state)
+        rval.trials.refresh()
+        for item in new_trials:
+            vals = item['misc']['vals']
+            parameter = dict()
+            for key in vals:
+                try:
+                    parameter[key] = vals[key][0].item()
+                except Exception:
+                    parameter[key] = None
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: log exception here and only catch expected Exception type.

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py (hunk 1 above)
Comment (leelaylay) [reply src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: I will fix it soon. Thank you.

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py (hunk 2 above)
Comment (leelaylay) [reply src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: It doesn't cost time and the time is about 3-5 µs.

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py (hunk 3 above)
Comment (leelaylay) [reply src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: This is the origin code and I think it reasonable.

**Comment by leelaylay** [comment]
Sorry, the origin API of hyperopt does not support multiple trials for tpe and anneal.

**Pull request closed without merging**